exp/api/v1beta2/rosamachinepool_types.go

```diff
@@ -33,6 +33,13 @@ type RosaMachinePoolSpec struct {
 	// +kubebuilder:validation:Pattern:=`^[a-z]([-a-z0-9]*[a-z0-9])?$`
 	NodePoolName string `json:"nodePoolName"`
 
+	// Version specifies the penshift version of the nodes associated with this machinepool.
+	// ROSAControlPlane version is used if not set.
+	//
+	// +optional
+	// +kubebuilder:validation:Pattern:=`^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`
```

Comment:
Do we have some provenance for this regex? Does it faithfully capture all possible semantic versions?

Also, please prefer to add code-gen comments above GoDoc so they do not get added to the user-facing documentation:

```go
	// +optional
	// +kubebuilder:validation:Pattern:=`^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`

	// Version specifies the penshift version of the nodes associated with this machinepool.
	// ROSAControlPlane version is used if not set.
	Version string `json:"version,omitempty"`
```

Reply:
https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

Reply:
See https://github.com/kubernetes-sigs/cluster-api/blob/main/util/version/version.go instead of inlined regex?

Reply:
are you suggesting we add a validation webhook that uses this to validate the version? or is there a way to reference the regex variables in `+kubebuilder:validation:Pattern`?

Reply:
Yeah correct

Reply:
added a webhook